client/prompt.lua

```diff
@@ -26,25 +25,20 @@ local function startPrompt(promptTable)
 
   promptIsOpen = true
 
-  RegisterNuiCallbackType(cbStr)
-  local eventData = AddEventHandler('__cfx_nui:' .. cbStr, function(data, cb)
-    p:resolve(data)
+  RegisterRawNuiCallback(cbStr, function(data, cb)
+    local resp = json.decode(data.body)
+
+    p:resolve(resp)
     promptIsOpen = false
-    cb({})
+    cb({ body = json.encode({}) })
```

Comment:
Since I missed this the first time I reviewed

```suggestion
    cb({ body = '{}' })
```
Should eliminate unnecessary encode call.

Reply:
Not sure why Dunak chose to use RegisterRawNuiCallback in the first place actually, The other week when working on we found that unregistering the raw nui callback would for some reason a trigger completely unrelated event 🤷

Reply:
RegisterRawNuiCallback is being used in conjunction with UnregisterRawNuiCallback to eliminate possible memory leaky behavior in the underlying ResourceUI container. 

You can see the following for more information.
https://github.com/citizenfx/fivem/pull/1045